Code/Framework/AzCore/Tests/Platform/Common/UnixLike/Tests/UtilsTests_UnixLike.cpp

```diff
@@ -36,4 +36,22 @@ namespace UnitTest
         ASSERT_TRUE(absolutePath);
         EXPECT_STRCASEEQ(executableDirectory, absolutePath->c_str());
     }
+
+    TEST_F(UtilsUnixLikeTestFixture, SetEnv_GetExpectedVariable_Succeeds)
+    {
+        AZ::Utils::SetEnv("UT-O3DE-ENVVAR", "TEST", 1);
+
+        auto result = std::getenv("UT-O3DE-ENVVAR");
+        ASSERT_TRUE(result != nullptr);
```

Comment:
For checking conditions in Googletest, the relational binary predicate macros should be used when checking non-boolean values as it provides better failure messaging as the variables can be resolved: https://github.com/google/googletest/blob/main/docs/advanced.md#predicate-assertions-for-better-error-messages

```suggestion
        ASSERT_NE(nullptr, result);
```

Using ASSERT_TRUE/EXPECT_TRUE can only print the condition 
```
Value of: result !=nullptr
Actual: false 
Expected: true
```
instead of a failure message that provides better information
such 
```
Expected: nullptr
Actual: <Memory Address>
```

Reply:
Moved utils to AzTest